### PR TITLE
lightning: Compress Reader/Writer supports reading/writing Snappy/Zstd type compressed files

### PR DESCRIPTION
**Pull request by lyzx2001**
### What problem does this PR solve?

Issue Number: ref #38514 

Problem Summary:

### What is changed and how it works?
Compress Reader supports reading Snappy/Zstd type compressed files
Compress Writer supports writing Snappy/Zstd type compressed files

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

**Comment by hawkingrei** [comment]
BTW, Have we a plan to support zstd ?

**Comment by lyzx2001** [comment]
> BTW, Have we a plan to support zstd ?
Thanks for the suggestion. I will try to implement supporting zstd in the later PR if possible.

**Comment by lyzx2001** [comment]
> BTW, Have we a plan to support zstd ?

Thanks for the suggestion. I will try to implement supporting zstd in the later PR if possible.

**Review by lance6716 — APPROVED** [review]

**Comment by lichunzhu** [comment]
snappy part LGTM. Looking forward to lzo/zstd.

**Comment by lyzx2001** [comment]
> snappy part LGTM. Looking forward to lzo/zstd.

Okay, in progress

File: br/pkg/storage/writer_test.go [hunk 1]
```diff
@@ -135,7 +134,49 @@ func TestCompressReaderWriter(t *testing.T) {
...
+		// test withCompression Open
+		r, err = storage.Open(ctx, fileName)
+		require.NoError(t, err)
+		content, err := io.ReadAll(r)
+		require.NoError(t, err)
+		require.Equal(t, strings.Join(test.content, ""), string(content))
+
+		require.Nil(t, file.Close())
+	}
+	compressTypeSnappyArr := []CompressType{Snappy}
```
Comment (lichunzhu) [inline br/pkg/storage/writer_test.go:178]: Can we combine these two functions?

Comment (lyzx2001) [reply br/pkg/storage/writer_test.go]: Modified

File: br/pkg/storage/writer_test.go [hunk 2]
```diff
@@ -102,8 +102,15 @@ func TestCompressReaderWriter(t *testing.T) {
 		ctx := context.Background()
 		storage, err := Create(ctx, backend, true)
 		require.NoError(t, err)
-		storage = WithCompression(storage, Gzip)
-		fileName := strings.ReplaceAll(test.name, " ", "-") + ".txt.gz"
+		storage = WithCompression(storage, test.compressType)
+		var suffix string
+		if test.compressType == Gzip {
```
Comment (lichunzhu) [inline br/pkg/storage/writer_test.go:107]: How about implementing a `String()` function for these compressType?

Comment (lyzx2001) [reply br/pkg/storage/writer_test.go]: Modified

**Comment by lyzx2001** [comment]
zstd is supported now.

**Review by lichunzhu — APPROVED** [review]
LGTM

**Pull request merged**